### PR TITLE
Updated some variables and methods inside summary-page

### PR DESCRIPTION
**Pull request by joanofdart**
Used the time today to work on the recursive function, but I'm not accomplishing the expected behavior.
I'm pushing some changes I did while working on the function.

### CODE REVIEW

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 1]
```diff
@@ -331,25 +331,23 @@ <h6>
...
-                </strong>
-                <span>&nbsp;</span>
-                <span class="text-primary">
-                  <missing-control [control]="review.get('revueLink')" [link]="getPath('reviews')" fragment="reviews"></missing-control>
-                </span>
-                "<missing-control [control]="review.get('criticQuote')" [link]="getPath('reviews')" fragment="reviews"></missing-control>"
-              </ng-template>
-            </li>
+          <ng-container *ngIf="!form.review.controls.length else displayReviews">
+            <a [routerLink]="getPath('reviews')" fragment="festival"class="missing-control">
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:335]: missing space

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 2]
```diff
@@ -331,25 +331,23 @@ <h6>
...
-                  <missing-control [control]="review.get('journalName')" [link]="getPath('reviews')" fragment="reviews"></missing-control>
-                </strong>
-                <span>&nbsp;</span>
-                <span class="text-primary">
-                  <missing-control [control]="review.get('revueLink')" [link]="getPath('reviews')" fragment="reviews"></missing-control>
-                </span>
-                "<missing-control [control]="review.get('criticQuote')" [link]="getPath('reviews')" fragment="reviews"></missing-control>"
-              </ng-template>
-            </li>
+          <ng-container *ngIf="!form.review.controls.length else displayReviews">
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:334]: I think it'd be better to switch this around.
`*ngIf="form.review.controls.length else noReviews"`

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 3]
```diff
@@ -676,6 +677,24 @@ <h6>
               {{ form.promotional.get('screener_link').value }}
             </ng-template>
           </li>
+          <li>
+            <mat-icon svgIcon="eye" color="primary"></mat-icon><strong>Other Links </strong>
+            <ng-container *ngIf="objectHasNoValue(otherLinks) else displayOtherLinks">
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:682]: I would also switch this one around

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts [hunk 4]
```diff
@@ -38,18 +38,18 @@ export class MovieFormSummaryComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit(): void {
+    console.log('this.form', this.form);
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.ts:41]: 💩

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts [hunk 5]
```diff
@@ -98,17 +98,17 @@ export class MovieFormSummaryComponent implements OnInit, OnDestroy {
...
           this.missingFields.push(field);
         }
-        if (control instanceof FormGroup) {
-          recursiveFunc(control);
-        } else if (control instanceof FormArray) {
+        if (control instanceof FormArray || control instanceof FormGroup) {
           recursiveFunc(control);
         }
       });
+      console.log('Fields', this.invalidFields, this.missingFields)
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.ts:105]: 💩

File: libs/movie/src/lib/movie/pages/summary/summary.component.ts [hunk 6]
```diff
@@ -98,17 +98,17 @@ export class MovieFormSummaryComponent implements OnInit, OnDestroy {
...
     }
     recursiveFunc(formToInvestigate);
   }
 
   public reviewHasNoValue(review: MovieReviewForm) {
+    console.log('Review', review);
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.ts:111]: 💩

File: libs/movie/src/lib/movie/pages/summary/summary.component.html [hunk 7]
```diff
@@ -331,25 +331,23 @@ <h6>
...
-            </li>
+          <ng-container *ngIf="!form.review.controls.length else displayReviews">
+            <a [routerLink]="getPath('reviews')" fragment="festival"class="missing-control">
+              Missing
+            </a>
           </ng-container>
+          <ng-template #displayReviews>
+            <li *ngFor="let review of form.review.controls">
+              <strong>
+                <missing-control [control]="review.get('journalName')" [link]="getPath('reviews')" fragment="reviews"></missing-control>
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/pages/summary/summary.component.html:342]: Create some getters on the review form for journalName (line 342), revueLink (line 346) and criticQuote (line 348)

**Review by RemcoSimonides — CHANGES_REQUESTED** [review]

File: libs/movie/src/lib/movie/pages/summary/summary.component.html (hunk 2 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/summary/summary.component.html:334]: We could, but this change is just a syntatic sugar to what we already have.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html (hunk 7 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/summary/summary.component.html:342]: No need : o I do create getters when we have something else to get from. But calling get here is fine.

File: libs/movie/src/lib/movie/pages/summary/summary.component.html (hunk 1 above)
Comment (RemcoSimonides) [reply libs/movie/src/lib/movie/pages/summary/summary.component.html:335]: `fragment="festival"class="missing-control"` still missing space between fragment en class ;)

**Review by RemcoSimonides — APPROVED** [review]

**Pull request merged**